airbyte-integrations/connectors/source-notion/source_notion/components.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from dataclasses import dataclass, field
+from typing import Any, List, Mapping, MutableMapping, Optional
+
+import pendulum
+from airbyte_cdk.sources.declarative.extractors.record_filter import RecordFilter
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation
+from airbyte_cdk.sources.declarative.types import StreamSlice, StreamState
+
+
+@dataclass
+class NotionUserTransformation(RecordTransformation):
+    """
+    Custom transformation that conditionally transforms Notion User records of type "bot",
+    only when the record contains additional nested "owner" info.
+    This transformation moves the data in the `owner.{owner_type}` field into a new `owner.info` field for clarity.
+    """
+
+    def transform(self, record: MutableMapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        owner = record.get("bot", {}).get("owner")
+        if owner:
+            owner_type = owner.get("type")
+            owner_info = owner.get(owner_type)
+            if owner_type and owner_info:
+                record["bot"]["owner"]["info"] = owner_info
+                del record["bot"]["owner"][owner_type]
+        return record
+
+
+@dataclass
+class NotionPropertiesTransformation(RecordTransformation):
+    """
+    Transforms the nested 'properties' object within a Notion Page/Database record into a more
+    normalized form. In Notion's API response, 'properties' is a dictionary where each key
+    represents the name of a property and its value contains various metadata and the property's
+    actual value.
+
+    The transformed 'properties' will consist of an array where each element is a dictionary
+    with two keys: 'name', holding the original property name, and 'value', containing the
+    property's content.
+    """
+
+    def transform(self, record: MutableMapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        properties = record.get("properties", {})
+        transformed_properties = [{"name": name, "value": value} for name, value in properties.items()]
+        record["properties"] = transformed_properties
+        return record
+
+
+@dataclass
+class NotionDataFeedFilter(RecordFilter):
```

Comment:
Why this class cannot be implemented in the low code? Is smth like this possible:
```
        record_filter:
          condition: "{{ record[parameters['cursor_field']] >= (stream_state.get(parameters['cursor_field'], config.get('start_date')) or "") }}"
```